src/containers/CallIn/ThankYou/ThankYou.js

```diff
@@ -57,35 +63,59 @@ class ThankYou extends Component {
         const homeDistrictNumber = getUrlParameter(params, 'd') || undefined;
         const trackingToken = getUrlParameter(params, 't') || undefined;
         const callerId = getUrlParameter(params, 'c') || undefined;
-        const callWasTracked = getUrlParameter(params, 'm') || false;
         this.removeTrackingGetArgs();
         this.fetchDistricts((districts) => {
             const calledDistrict = this.findDistrictByStateNumber(calledState, calledNumber, districts);
-            if(!calledDistrict && !calledDistrict.districtId){
+            if (!calledDistrict && !calledDistrict.districtId) {
                 this.setState({
                     statsError: Error("No district found")
                 })
                 return;
             } else {
-                this.setStats(calledDistrict);
                 const eligibleCallTargets = this.eligibleCallTargetDistrictIds(
-                    homeDistrictNumber, 
-                    calledState, 
-                    calledNumber, 
+                    homeDistrictNumber,
+                    calledState,
+                    calledNumber,
                     districts
-                )
+                );
                 this.setState({
-                    callWasTracked,
+                    calledState,
+                    calledNumber,
                     homeDistrictNumber,
                     trackingToken,
                     callerId,
                     district: calledDistrict,
-                    eligibleCallTargets: eligibleCallTargets.length ? eligibleCallTargets : null,                    
-                })
+                    eligibleCallTargets: eligibleCallTargets.length ? eligibleCallTargets : null,
+                });
+                this.reportCall()
```

Comment:
`setState` is async, so no guarantees that `reportCall()` has access to new state. Check out `this.setState({}, () => { ... })`